### PR TITLE
Add mutation for deleting a ClientAlert

### PR DESCRIPTION
**Pull request by martha**
[//]: # 'remove this if PR is for a release-* branch'
# _Please squash merge this PR_

## Description

This PR creates a graphql mutation enabling deletion of a client alert if the user has permission to do so, and rspec tests for the deletion mutation.

## Type of change
[//]: # 'remove options that are not relevant'
- [ ] Bug fix
- [x] New feature (adds functionality)
- [ ] Code clean-up / housekeeping
- [ ] Dependency update

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (rubocop)
- [x] I have updated the documentation (or not applicable)
- [x] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

**Comment by martha** [comment]
hm, I unthinkingly updated the base branch of this PR to be `release-103` but this was obviously a mistake since 102 seems to have contained a lot of changes that aren't on 103 yet. I'm wondering now whether I should revert that change or just wait for `release-102` to be merged into `release-103`?

**Comment by eanders** [comment]
@martha , I'm not sure on when this needs to be deployed.  102 will be going up to production next week, 103 the week after.  If this can wait two weeks, you should be fine (I just updated 102, and 103, so updating again should get everything else from 102.)  If it needs to get merged into 102, you can probably just switch the base.  103 doesn't have anything in it that isn't in 102 yet.

**Comment by martha** [comment]
thank you, @eanders !! changing the base branch to 102 at least makes the diff show the actual changes that this PR represents, so I'll keep it that way for now. I'm not sure if the change needs to go out next week or should wait two weeks, so I'll wait for Gig and/or Mary Ann to weigh in on that on Monday.

File: drivers/hmis/app/graphql/mutations/delete_client_alert.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+module Mutations
+  class DeleteClientAlert < CleanBaseMutation
+    argument :id, ID, required: true
+
+    field :client_alert, Types::HmisSchema::ClientAlert, null: true
+    field :errors, [Types::HmisSchema::ValidationError], null: false, resolver: Resolvers::ValidationErrors
+
+    def resolve(id:)
+      record = Hmis::ClientAlert.find(id)
+      raise 'Access denied' unless allowed?(permissions: [:can_manage_client_alerts])
```
Comment (gigxz) [inline drivers/hmis/app/graphql/mutations/delete_client_alert.rb]: we should specify the "entity" for the permissions here, which is the client

```suggestion
      raise 'Access denied' unless current_permission?(permission: :can_manage_client_alerts, entity: record.client)
```

**Comment by gigxz** [comment]
Lets keep it on 102 if that's alright. The rest of the Client Alert changes are there, so that will let us QA everything together.

This looks good to me after the permission check fix!

File: drivers/hmis/app/graphql/mutations/delete_client_alert.rb (hunk 1 above)
Comment (martha) [reply drivers/hmis/app/graphql/mutations/delete_client_alert.rb]: Thank you!

**Review by gigxz — APPROVED** [review]

**Pull request merged**